pkg/v1/tkr/controllers/source/tkr_source_controller.go

```diff
@@ -50,8 +50,8 @@ type reconciler struct {
 }
 
 // Reconcile performs the reconciliation step
-func (r *reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	ctx, cancel := context.WithCancel(r.ctx)
+func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) { //nolint:staticcheck
```

Comment:
@imikushin Please help review the changes wrt tkr-controller. Thanks

Reply:
Looks good so far. 

One question: why do we need `//nolint:staticcheck`? A comment with explanation wouldn't hurt.

Reply:
Removed the nolint comments and fixed the lint issues